src/explorer/webview.ts

```diff
@@ -211,15 +211,22 @@ class ExplorerWebview {
     private createExplorerHeaderContent(resultInfo: ResultInfo): HTMLDivElement {
         const header = this.createElement("div", { id: "title" }) as HTMLDivElement;
 
-        header.appendChild(this.createElement("label", { id: "titleruleid", text: resultInfo.ruleId }));
-        header.appendChild(this.createElement("label", { id: "titlerulename", text: resultInfo.ruleName }));
+        if (resultInfo.ruleId !== undefined || resultInfo.ruleName !== undefined) {
+            header.appendChild(this.createElement("label", { id: "titleruleid", text: resultInfo.ruleId }));
+            header.appendChild(this.createElement("label", { id: "titlerulename", text: resultInfo.ruleName }));
+        } else {
+            header.appendChild(this.createElement("label", { id: "titlerulename", text: "No Rule Info" }));
+        }
         header.appendChild(this.createElement("label", { text: " | " }));
-        if (resultInfo.locations[0] !== undefined) {
-            const filenameandline = resultInfo.locations[0].fileName + " (" +
+
+        let filenameandline = "No Location";
+        if (resultInfo.locations[0] !== null && resultInfo.locations[0] !== undefined) {
```

Comment:
>resultInfo.locations [](start = 12, length = 20)

can resultInfo.Locations itself ever be null?

Reply:
good question, no the location will always be defined as at least an empty array

---
In reply to: [217875015](https://github.com/Microsoft/sarif-vscode-extension/pull/94#discussion_r217875015) [](ancestors = 217875015)